### DISCUSSION TITLE
Removal of webdrivermanager

### DISCUSSION THREAD

**Discussion by kalupator** [eliasnogueira/selenium-java-lean-test-architecture#254] [General]
Hello!
Please tell why webdrivermanager was removed from the project? Now I get an error:

> java.lang.IllegalStateException: The path to the driver executable The path to the driver executable must be set by the webdriver.chrome.driver system property; for more information, see https://chromedriver.chromium.org/. The latest version can be downloaded from https://chromedriver.chromium.org/downloads

It's possible to fix if download the web driver for each browser again and setup path manually:
`System.setProperty("webdriver.chrome.driver", "c:\\driver\\chromedriver.exe");`

I tried to return webdrivermanager, but I get an error:
This version of ChromeDriver only supports Chrome version 114

**Comment by eliasnogueira** [comment]
Hi @kalupator,
The `webdrivermanager` library was removed in favor of [Selenium Driver Manager](https://www.selenium.dev/documentation/selenium_manager/), which has been embedded in the Selenium library since version 4.11.0.

You can add it back and use the previous approach set before the change used in the `BrowserFactory` class you can see in the following commits:
* https://github.com/eliasnogueira/selenium-java-lean-test-architecture/commit/3d4f90270be6b86c4fe722a75fdd408d983b2520
* https://github.com/eliasnogueira/selenium-java-lean-test-architecture/commit/740008ad387f08e8c251ee963c7ca1a49823861e
* https://github.com/eliasnogueira/selenium-java-lean-test-architecture/commit/1dbf635e0548cff8747ea04e5e45f6e2a461a83e

I can't help you with the error you are getting now because I don't know how you have implemented it.
I don't know if you did but don't forget to remove the fixed path to the driver when using either the `webdrivermanager` or the Selenium Driver Manager approach.

**Reply by kalupator** [reply]
Thanks for the answer!
I didn't use a fixed path to the webdriver because webdrivermanager downloaded the required webdriver itself.

Now I have updated to Selenium 4.18.1 and removed webdrivermanager, in the hope that the new Selenium will also download the required webdriver, as webdrivermanager did.

But I get the error described above that the webdriver was not found.

Perhaps I need to add some more dependencies or delete the cache of the old version of Selenium?

```
        <dependency>
            <groupId>org.seleniumhq.selenium</groupId>
            <artifactId>selenium-java</artifactId>
            <version>4.18.1</version>
        </dependency>
```

I can't return webdrivermanager because apparently it no longer downloads the Chrome web driver above version 114.

**Reply by eliasnogueira** [reply]
I just checked the ChromeDriver page and I saw a mention of Chrome version 115 or newer
https://chromedriver.chromium.org/downloads

I'm running it from my local machine using the currently implemented approach with Selenium Driver Manager, and I can see the following in the `~/.cache/selenium`

**resolution.properties**
```
#WebDriverManager Resolution Cache
#Wed Feb 21 20:50:30 CET 2024
chrome=121
chrome-ttl=21\:50\:30 21/02/2024 CET
chrome121=public
chrome121-ttl=20\:50\:30 22/02/2024 CET
```

The following list of the available chrome driver at `.cache/selenium/chromedriver/mac-arm64`
```
120.0.6099.109
121.0.6167.184
121.0.6167.85 
122.0.6261.128
123.0.6312.86
```

My local Chrome browser version is `123.0.6312.59`
So, it's solving correctly the version.

Could you share the same info, so we can double-check and see the possible root cause?

**Reply by kalupator** [reply]
My local Chrome browser version is 123.0.6312.59.
In folder `~/.cache/selenium` the last downloaded webdriver version was 114.
I deleted the entire catalog `~/.cache/selenium` and the new version simply won't download.
Apparently this is the problem. I need to understand why Selenium does not start searching and downloading the webdriver.

Stacktrace after execute line - `return new ChromeDriver(getOptions())` in createLocalDriver():
```
java.lang.IllegalStateException: The path to the driver executable The path to the driver executable must be set by the webdriver.chrome.driver system property; for more information, see https://chromedriver.chromium.org/. The latest version can be downloaded from https://chromedriver.chromium.org/downloads

	at org.openqa.selenium.internal.Require$StateChecker.nonNull(Require.java:311)
	at org.openqa.selenium.remote.service.DriverService.findExecutable(DriverService.java:135)
	at org.openqa.selenium.chrome.ChromeDriverService.access$000(ChromeDriverService.java:39)
	at org.openqa.selenium.chrome.ChromeDriverService$Builder.findDefaultExecutable(ChromeDriverService.java:233)
	at org.openqa.selenium.remote.service.DriverService$Builder.build(DriverService.java:437)
	at org.openqa.selenium.chrome.ChromeDriverService.createServiceWithConfig(ChromeDriverService.java:142)
	at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:70)
```

**Reply by kalupator** [reply]
I found the reason, it's a matter of dependencies:
`<artifactId>selenium-java</artifactId>` updated to 4.18.1
but for some reason the rest of the dependencies remain 4.5.3

I haven't found the reason for this yet.

![Screenshot 2024-03-28 233946](https://github.com/eliasnogueira/selenium-java-lean-test-architecture/assets/155086518/ba563a15-0617-486d-bcda-8f09a881b27d)

I only have one dependency for selenium:
```
        <dependency>
            <groupId>org.seleniumhq.selenium</groupId>
            <artifactId>selenium-java</artifactId>
            <version>4.18.1</version>
        </dependency>
```

**Reply by eliasnogueira** [reply]
I am happy that you could solve the problem.
Sometimes, IntelliJ caches the dependencies and it's important to reload through the IDE as well.
